io/io/inc/ROOT/RRawFile.hxx

```diff
@@ -117,42 +108,32 @@ private:
       size_t CopyTo(void *buffer, size_t nbytes, std::uint64_t offset);
    };
    /// To be used modulo kNumBlockBuffers, points to the last used block buffer in fBlockBuffers
-   unsigned int fBlockBufferIdx;
+   unsigned int fBlockBufferIdx = 0;
    /// An active buffer and a shadow buffer, which supports "jumping back" to a previously used location in the file
    RBlockBuffer fBlockBuffers[kNumBlockBuffers];
    /// Memory block containing the block buffers consecutively
-   unsigned char *fBufferSpace;
+   std::unique_ptr<unsigned char[]> fBufferSpace;
    /// The cached file size
-   std::uint64_t fFileSize;
+   std::uint64_t fFileSize = kUnknownFileSize;
    /// Files are opened lazily and only when required; the open state is kept by this flag
-   bool fIsOpen;
+   bool fIsOpen = false;
 
 protected:
    std::string fUrl;
    ROptions fOptions;
    /// The current position in the file, which can be changed by Seek, Read, and Readln
-   std::uint64_t fFilePos;
+   std::uint64_t fFilePos = 0;
 
-   /**
-    * OpenImpl() is called at most once and before any call to either DoReadAt or DoGetSize. If fOptions.fBlocksize
-    * is negative, derived classes are responsible to set a sensible value. After a call to OpenImpl(),
-    * fOptions.fBlocksize must be larger or equal to zero.
-    */
+   /// OpenImpl() is called at most once and before any call to either DoReadAt or DoGetSize. If fOptions.fBlocksize
+   /// is negative, derived classes are responsible to set a sensible value. After a call to OpenImpl(),
+   /// fOptions.fBlocksize must be larger or equal to zero.
    virtual void OpenImpl() = 0;
-   /**
-    * Derived classes should implement low-level reading without buffering. Short reads indicate the end of the file,
-    * therefore derived classes should return nbytes bytes if available.
-    */
+   /// Derived classes should implement low-level reading without buffering. Short reads indicate the end of the file,
+   /// therefore derived classes should return nbytes bytes if available.
    virtual size_t ReadAtImpl(void *buffer, size_t nbytes, std::uint64_t offset) = 0;
-   /// Derived classes should return the file size or kUnknownFileSize
```

Comment:
We probably should make `kUnknownFileSize` private and adjust its comment accordingly (used internally to as a marker if the file size was already queried).